### ISSUE TITLE
Suggestion: Add the fields instead of adding the technologies randomly.

### ISSUE THREAD

**Issue by ibilalkayy** [loftwah/linux-for-pirates#23]
Hey Dean, Why don't you deliver the content based on different fields like web development, DevOps, mobile development, etc? If someone wants to learn a specific field then there should be a link to that field.

A question may arise if some content is common in multiple fields then should we write them in all the fields?

**Answer:** No we should include all the common content in a separate file and link them in the fields that are using them.

What do you think about it? Do you have any idea of managing these technologies in a proper way? I have created a repository for that. You can [check it](https://github.com/ibilalkayy/full-path) out.

**Comment by loftwah** [comment]
This makes sense now, and I like it. If you want to do this, I will accept the PR, but it will impact GitHub pages. It may be worth doing this after mdbook is in place, or we could look for another solution (or keep the current one).

**Comment by ibilalkayy** [comment]
No problem. Whenever you want, I will make a PR.

**Comment by loftwah** [comment]
Cheers, let me think about it and I'll see if I can give `mdbook` another go. Once that is ready we can do whatever we want and it won't matter. At the moment I only have `index.md` working.

**Comment by ibilalkayy** [comment]
No problem. I am looking to collaborate more. What are the next things to do about it? I noticed that you have the same definition under multiple headings. Why?

**Comment by loftwah** [comment]
Because I didn't know where to put it, and if I didn't write it down at the time, I could have forgotten about it.

If you like, you could be in charge of the structure. You seem to know quite a bit about how it should work, and you've done this before. I am happy to assign it to you, and I am not phased if you break GitHub pages. We can redo that later.

It probably wouldn't be difficult to fix, I'm just not sure how quickly I'll be able to get to it.

**Comment by ibilalkayy** [comment]
Thank you, @loftwah for assigning me! I will take a look at it.

**Comment by loftwah** [comment]
I just merged your latest request. I'll translate it and get you to look over it again before I merge any of those changes.

From here I think we should start only merging changes based on PRs (I know you do, but I've been pushing straight to main, which isn't great).

Appreciate it.

**Comment by loftwah** [comment]
I think this may break GitHub Pages but it isn't a big deal and I can work out how to fix it later.

**Comment by loftwah** [comment]
It handles it surprisingly well, wow. Well done.

Some of the images are broken now so we'll need to fix it. We should raise another issue (I can do it later if no one else does).
